.changeset/weak-zebras-pull.md

```diff
@@ -0,0 +1,21 @@
+---
+"xstate": minor
+---
+
+Removed the ability to pass a model as a generic to `createMachine`, in favour of `model.createMachine`. This lets us cut an overload from the definition of `createMachine`, meaning errors become more targeted and less cryptic.
```

Comment:
This was already being implemented in https://github.com/statelyai/xstate/pull/2518/files#diff-00f777398251992f42d7061fc6a79e0c6e7c8aebfd257cb1d8c04129c84e0273R51 and accepted back then so I think it's an OK change.